ddl/column_type_change_test.go

```diff
@@ -552,13 +552,14 @@ func (s *testColumnTypeChangeSuite) TestColumnTypeChangeFromStringToOthers(c *C)
 	tk.MustExec("insert into t values ('123.45', '123.45', '123.45', '123.45', '123.45', '123.45', '123', '123')")
 	tk.MustExec("alter table t modify c decimal(7, 4)")
 	tk.MustExec("alter table t modify vc decimal(7, 4)")
-	tk.MustExec("alter table t modify bny decimal(7, 4)")
+	// alter binary 0x3132332E34350000 to decimal(7,4) will get error
+	tk.MustGetErrCode("alter table t modify bny decimal(7, 4)", mysql.ErrTruncatedWrongValue)
```

Comment:
We change existing test case here. Does it mean this PR changes TiDB behavior explicitly (more than a warning)?

Reply:
Yes, In MySQL,  this case will directly return error